ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/RoutingResolveContext.kt

```diff
@@ -170,6 +175,7 @@ public class RoutingResolveContext(
         val finalResolve = resolveResult
 
         if (finalResolve.isEmpty()) {
+            LOGGER.trace("No route matched for ${call.request.uri}")
```

Comment:
why do we need both logging and tracing in routing? can they be merged?

Reply:
Tracing has more information and is better structured. Maybe we can enable it automatically, if log level is trace?

Reply:
Done!